### PR TITLE
Replace Calculus.jl with DiffEqDiffTools functionality for finite differencing

### PR DESCRIPTION
**Pull request by dextorious**
This should only be merged after DiffEqDiffTools gets tagged and @ChrisRackauckas is happy with the changes overall.

### CODE REVIEW

File: src/integrators/rosenbrock_integrators.jl [hunk 1]
```diff
@@ -31,7 +31,7 @@ end
       if alg_autodiff(integrator.alg)
         ForwardDiff.derivative!(dT, tf, vec(du2), t)
       else
-        dT = Calculus.finite_difference(tf, t, integrator.alg.diff_type)
+        DiffEqDiffTools.finite_difference!(dT, tf, t, Val{integrator.alg.diff_type}, vec(du2), Val{:DiffEqDerivativeWrapper})
```
Comment (ChrisRackauckas) [inline src/integrators/rosenbrock_integrators.jl]: `integrator.alg.diff_type` should already be stored as a `Val`

Comment (dextorious) [inline src/integrators/rosenbrock_integrators.jl]: Sorry about that, leftover from testing. Fixed.

File: src/algorithms.jl [hunk 2]
```diff
@@ -121,15 +121,15 @@ struct StrangSplitting <: OrdinaryDiffEqAlgorithm end
 
 struct ImplicitEuler{CS,AD,F,K,T,T2,Controller} <: OrdinaryDiffEqNewtonAdaptiveAlgorithm{CS,AD,Controller}
   linsolve::F
-  diff_type::Symbol
+  diff_type::DataType
```
Comment (ChrisRackauckas) [inline src/algorithms.jl]: Not type-stable.

Comment (dextorious) [inline src/algorithms.jl]: Hopefully fixed in the latest commit via parametrization.

**Pull request merged**